file/builder.go

```diff
@@ -51,15 +53,11 @@ func (b *stateBuilder) build() (*utils.KongRawState, *utils.KonnectRawState, err
 		return nil, nil, err
 	}
 
-	// defaulter
-	var kongDefaults KongDefaults
-	if b.targetContent.Info != nil {
-		kongDefaults = b.targetContent.Info.Defaults
-	}
-	b.defaulter, err = defaulter(kongDefaults)
+	defaulter, err := defaulter(b.ctx, b.client, b.targetContent, b.isKonnect)
```

Comment:
Not in scope but we really need to pas the defaulter as an input to rather than constructing it here.

Reply:
The only reason I did it so is to avoid duplicating the same exact lines for Kong and Konnect